uint8 INSERT_AT=0
uint8 INSERT_BEFORE=1
uint8 INSERT_AFTER=2
uint8 REMOVE=3

uint8    call_mode
string   reference_tag
string   tag
string   operation_type
string[] parameters